### PR TITLE
authz: filter out soft-deleted repositories when syncing permissions

### LINKED ISSUE #9076: Filter out soft-deleted repositories when syncing permissions
We do eventually skip them in our current implementation, but it would be nice avoid returning them in the first place.

For both `PermsStore.RepoIDsWithNoPerms` and `PermsStore.ReposIDsWithOldestPerms` methods, they should filter out soft-deleted repos.

### PR DESCRIPTION
**Pull request by unknwon**
This PR makes our `PermsStore.RepoIDsWithNoPerms` and `PermsStore.ReposIDsWithOldestPerms` methods to filter out soft-deleted repos, otherwise we'll always get these deleted repos as repos have oldest permissions because their permissions in fact aren't getting updated.

Fixes #9076

### CODE REVIEW

File: enterprise/cmd/frontend/db/perms_store.go [hunk 1]
```diff
@@ -165,50 +165,50 @@ func (s *PermsStore) SetUserPermissions(ctx context.Context, p *authz.UserPermis
 	// Load stored object IDs of both added and removed.
 	changedIDs := roaring.Or(added, removed).ToArray()
 
-	// In case there is nothing to add or remove.
```
Comment (unknwon) [inline enterprise/cmd/frontend/db/perms_store.go]: Suggest split diff view.

File: enterprise/cmd/frontend/db/perms_store.go [hunk 2]
```diff
@@ -1345,7 +1345,8 @@ func (s *PermsStore) RepoIDsWithNoPerms(ctx context.Context) ([]api.RepoID, erro
 	q := sqlf.Sprintf(`
 -- source: enterprise/cmd/frontend/db/perms_store.go:PermsStore.RepoIDsWithNoPerms
 SELECT repo.id, '1970-01-01 00:00:00+00'::timestamptz FROM repo
-WHERE repo.private = TRUE
+WHERE repo.deleted_at IS NULL
+AND repo.private = TRUE
```
Comment (keegancsmith) [inline enterprise/cmd/frontend/db/perms_store.go]: this looks like the important change. Why did so much else change in this file?

Comment (unknwon) [reply enterprise/cmd/frontend/db/perms_store.go]: HAHA, sorry, I kinda mixed three things together.

1. Ignore soft-deleted repos
2. Always update `UpdatedAt` timestamp
3. Fix a test flaky

Found all during my e2e testing so committed together.

**Comment by unknwon** [comment]
I did Bitbucket Server e2e testing based on this PR, thanks everyone, merging :)

**Pull request merged**